src/main/scala/metadata.scala

```diff
@@ -0,0 +1,83 @@
+package suffuse
+
+/** AKey is an attribute key: a Metadata map key which carries
+ *  both the type of the corresponding value, and a value to
+ *  provide when there is no value in the map. This as opposed
+ *  to wrapping everything in sight in Option.
+ */
+final class AKey[A](description: String, empty: => A) {
```

Comment:
Having gotten lost in the code for a second (with all the interleaving inline comments) I have to say I had to go back to the definition (and scaladoc) to remind myself what `AKey` was. So perhaps this could be called `AttrKey` or maybe even `AttributeKey`.

I only mention it in passing, it's textbook bike-shedding on my part, but it might be interesting anecdotal evidence.